### ISSUE TITLE
Event params' type in invoked handlers are not the same as declared in event signature

### ISSUE THREAD

**Issue by atom-yang** [graphprotocol/graph-ts#151]
Related Contract: https://etherscan.io/address/0x3f966fa1c0606e19047ed72068d2857677e07ef4#code
Related Event: `event Harvested(address, address, uint256)`
Related Code: 
```typescript
export function handleNewHarvest(event: NewHarvestedEvent): void {
  // output: handle new harvest 0 type 4
  log.debug('handle new harvest 0 type ' + event.parameters[0].value.kind.toString(), []);
  // output: handle new harvest 1 type 4
  log.debug('handle new harvest 1 type ' + event.parameters[1].value.kind.toString(), []);
  // output: handle new harvest 2 type 4
  log.debug('handle new harvest 2 type ' + event.parameters[2].value.kind.toString(), []);
  // throw an error
  let profitTokenAddress = changetype<Address>(event.parameters[1].value.data as u32);
 // throw an error, related to https://github.com/graphprotocol/graph-ts/blob/fdd8f835316c68c66eb9fd9917c2c5c8f07bf904/chain/ethereum.ts#L38, since param0's ValueKind is 4, which is uint
  event.params.param0;
```
## Problem
in the event `Harvested(address, address, uint256)`, the param0 and param1's type is `Address`, but in the process of scanning, when trigger the callback function `handleNewHarvest()`, the params' type become `uint`, not Address, so I can't use `event.params.param0`, since it calls the method `toAddress()`, which has an assert.

So I tried to copy the code you convert the original data, but it still failed, probably the original data has been cut since different type.

Wish to get your support, thanks.

**Comment by leoyvens** [comment]
We need a link to the subgraph source code that reproduces this, and the error log when the assert is hit.

**Comment by atom-yang** [comment]
Ok, here is the github [repo](https://github.com/SashimiProject/sashimi-deradar-graph) and [thegraph](https://thegraph.com/explorer/subgraph/sashimiproject/sashimi-farm?selected=logs)
And sorry, for now, the callback of event doesn't output any debug log and the scanning process went fatal without anything tips.

**Comment by atom-yang** [comment]
> We need a link to the subgraph source code that reproduces this, and the error log when the assert is hit.

Ok, I redeploy this subgraph. And the debug output and error message went back, check this [subgraph]
Check the file `src/mappings/investment.ts` in my repo

(https://thegraph.com/explorer/subgraph/sashimiproject/sashimi-farm?selected=logs)
![WX20210114-102507](https://user-images.githubusercontent.com/52150262/104536311-d2ddf280-5652-11eb-9b28-2e1f441eb766.png)

**Comment by atom-yang** [comment]
Anyone can help? @leoyvens

**Comment by leoyvens** [comment]
Hi sorry for dropping the ball here. Have you tried checking if `event.parameters[1].value.kind` is not `ADRESS`, and in that case avoiding accessing `params.param1`?

**Comment by atom-yang** [comment]
Yeah, that's what I mentioned above, the type of event params is `Address`, but the  `event.parameters[1].value.kind` is uint, it makes the convert step failed in this file https://github.com/graphprotocol/graph-ts/blob/c18868afe9666a2283cbaa54692ea78608e8a50d/index.ts#L779

**Comment by leoyvens** [comment]
@atom-yang I meant to ask, is it always `UINT` in all events, or does it vary?

**Comment by atom-yang** [comment]
Not in all events, just in this event `Harvested`. Pls check the [repo](https://github.com/SashimiProject/sashimi-deradar-graph/blob/demo/subgraph.yaml) and [graph](https://thegraph.com/explorer/subgraph/sashimiproject/sashimi-farm?selected=logs)

**Comment by leoyvens** [comment]
@atom-yang I've gotten around to properly investigating this. The problem is that the event parameters don't have a name. Please work around this by editing the ABI to have different names for each parameter, such as:
```json
  {
    "anonymous": false,
    "inputs": [
      {
        "indexed": false,
        "internalType": "address",
        "name": "a",
        "type": "address"
      },
      {
        "indexed": false,
        "internalType": "address",
        "name": "b",
        "type": "address"
      },
      {
        "indexed": false,
        "internalType": "uint256",
        "name": "c",
        "type": "uint256"
      }
    ],
    "name": "Harvested",
    "type": "event"
  },
```

This is most likely a bug in ethabi, but I can't give an ETA on a fix so please use this workaround for now.
